github/messages.go

```diff
@@ -117,3 +147,44 @@ func validateSignature(signature string, payload, secretKey []byte) error {
 	}
 	return nil
 }
+
+// getWebhookEvent parses the payload and request, and build the Event.
+func getWebhookEvent(messageType string, payload []byte) (interface{}, error) {
+	eventType, ok := eventTypeMapping[messageType]
+	if !ok {
+		return nil, fmt.Errorf("unknown X-Github-Event in message: %v", messageType)
+	}
+
+	event := Event{
+		Type:       &eventType,
+		RawPayload: (*json.RawMessage)(&payload),
+	}
+	return event.Payload(), nil
+}
+
+// ParseWebhook parses the request, validates the signature, and returns
+// an interface to an activity event type (as returned by Event.Payload())
+//
+// secretKey is the GitHub Webhook secret message.
+//
+// Example usage:
+//
+//     func (s *GitHubEventMonitor) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+//       event, err := github.ParseWebHook(r, s.webhookSecretKey)
+//       if err != nil { ... }
+//       switch event := event.(type) {
+//       case CommitCommentEvent:
+//           processCommitCommentEvent(event)
+//       case CreateEvent:
+//           processCreateEvent(event)
+//       ...
+//       }
+//     }
+//
+func ParseWebHook(r *http.Request, secretKey []byte) (interface{}, error) {
```

Comment:
Very valid comment, thanks for the feedback.

To go even further on your side, passing the secret key to `ParseWebHook` may be confusing. I guess I'll keep the two functions seperated from each other.